support/k3d/k3d.go

```diff
@@ -0,0 +1,399 @@
+/*
```

Comment:
This should be in third-party.
I see that kwok in that package, but it should have also been in third-party.
package support is a place for API definitions for third-party support.
`kind` was supposed to be an example implementation for others (maybe it should have been in third-party as well)

Reply:
Good point.. I can move them all. But we have to retain some sort of aliasing to aid the import backward compatibility.. At least for kwok and the kind that we support today in an available release.. Right? 


If you are ok with that, I will take care of the required package refactoring

Reply:
Hi @harshanarayana if type aliasing will help maintain backward compat, then do it. 
At the very least, we should just move kwok.  Thanks for doing it.